opm/output/eclipse/Summary.cpp

```diff
@@ -102,18 +103,51 @@ measure rate_unit< rt::gas >() { return measure::gas_surface_rate; }
 template<> constexpr
 measure rate_unit< Phase::GAS >() { return measure::gas_surface_rate; }
 
-template< rt phase >
+template< rt phase, bool injection = true >
 inline quantity rate( const fn_args& args ) {
     double sum = 0.0;
 
     for( const auto* ecl_well : args.ecl_wells ) {
-        if( args.wells.wells.count( ecl_well->name() ) == 0 ) continue;
-        sum += args.wells.at( ecl_well->name() ).rates.get( phase, 0.0 );
+        const auto& name = ecl_well->name();
+        if( args.wells.wells.count( name ) == 0 ) continue;
+        const auto v = args.wells.at( name ).rates.get( phase, 0.0 );
+        if( ( v > 0 ) == injection )
+            sum += v;
     }
 
+    if( !injection ) sum *= -1;
     return { sum, rate_unit< phase >() };
 }
 
+template< rt phase, bool injection = true >
+inline quantity crate( const fn_args& args ) {
```

Comment:
Yeah. I mean, we can throw in an assert I suppose, but the big reason for just reusing everything was that wells already come in a list, so it would be zero or one iteration. Could probably use less noise though, I agree with that, so I'll look into it.